distributed/tests/test_worker.py

```diff
@@ -3351,6 +3351,78 @@ async def test_gather_dep_no_longer_in_flight_tasks(c, s, a):
         assert not any("missing-dep" in msg for msg in f2_story)
 
 
+@gen_cluster(client=True, nthreads=[("", 1)])
+async def test_gather_dep_cancelled_error(c, s, a):
+    """Something somewhere in the networking stack raises CancelledError while
+    gather_dep is running
+
+    See Also
+    --------
+    test_get_data_cancelled_error
+    https://github.com/dask/distributed/issues/8006
+    """
+    async with BlockedGetData(s.address) as b:
+        x = c.submit(inc, 1, key="x", workers=[b.address])
+        y = c.submit(inc, x, key="y", workers=[a.address])
+        await b.in_get_data.wait()
+        tasks = {
+            task for task in asyncio.all_tasks() if "gather_dep" in task.get_name()
+        }
+        assert tasks
+        # There should be only one task but cope with finding more just in case a
+        # previous test didn't properly clean up
+        for task in tasks:
+            task.cancel()
```

Comment:
What I do not like about this PR is this test. This test is just hooking into all tasks and are artificially cancelling those. The only place where this is in fact happening is during [`BaseWorker.close`](https://github.com/dask/distributed/blob/24bd9a05cf8f8dfa3d1aea502b8f11f48b37bf99/distributed/worker_state_machine.py#L3780-L3784) but this is clearly not what's concerning us, is it?

How else do we end up in this state?